src/main/java/org/takes/misc/Select.java

```diff
@@ -0,0 +1,159 @@
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.misc;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * Select elements into a new iterable with given condition.
+ *
+ * @author Jason Wong (super132j@yahoo.com)
+ * @version $Id$
+ * @since 0.13.8
+ */
+public final class Select<T> implements Iterable<T> {
+
+    /**
+     * Internal storage to hold the elements from iterables.
+     */
+    private final transient Iterable<T> list;
+
+    /**
+     * The condition to filter the element in the iterator.
+     */
+    private final transient Condition<T> condition;
+
+    /**
+     * To produce an iterable collection, determined by condition, combining a
+     * and b, with order of the elements in a first.
+     * @param itb Iterable to select
+     * @param cond To determine which element to add in the final iterable
+     */
+    public Select(final Iterable<T> itb, final Condition<T> cond) {
+        this.list = itb;
+        this.condition = cond;
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return new SelectIterator<T>(
+            this.list.iterator(),
+            this.condition
+        );
+    }
+
+    /**
+     * The select iterator to traverse the input iterables and return the
+     * elements from the list with given condition.
+     */
+    private static class SelectIterator<E> implements Iterator<E> {
+
+        /**
+         * The index pointing to the current element list.
+         */
+        private static final transient int HEAD = 0;
+
+        /**
+         * The iterator to reflect the traverse state.
+         */
+        private final transient Iterator<E> iterator;
+
+        /**
+         * The condition to filter the elements in the iterator.
+         */
+        private final transient Condition<E> condition;
+
+        /**
+         * The list storing the current object of the iterator.
+         */
+        private final transient List<E> current = new ArrayList<E>(1);
+
+        /**
+         * Ctor. ConcatIterator traverses the element.
+         * @param itr Iterator of the original iterable
+         * @param cond Condition to filter out elements
+         */
+        public SelectIterator(final Iterator<E> itr, final Condition<E> cond) {
+            this.condition = cond;
+            this.iterator = itr;
+            if (this.iterator.hasNext()) {
```

Comment:
@yegor256, this is to initialise the first element for ```hasNext()``` and ```next()``` to work with and it is needed as part of iterator initialisation even though no one called ```.next()```